assets/js/custom/Translation.js

```diff
@@ -74,17 +74,25 @@ class Translation {
     )
   }
 
-  setTranslationCredit (creditContainers, data) {
-    $(creditContainers.before).text(this.translatedByLineMap.before.replace('%provider%', this.providerMap[data.provider]))
-    $(creditContainers.between).text(this.translatedByLineMap.between.replace('%provider%', this.providerMap[data.provider]))
-    $(creditContainers.after).text(this.translatedByLineMap.after.replace('%provider%', this.providerMap[data.provider]))
+  setTranslationCredit (data, commentId) {
+    let creditSection = '#program'
+    let creditIdentifier = ''
+
+    if (commentId !== null) {
+      creditSection = '#comment'
+      creditIdentifier = '-' + commentId
+    }

```

Comment:
I think this is worse than the container approach you had before, because now it handles program translation and comment translation differently.

My suggestion was to define a new data structure for the container. So it's the same idea as the container, but just a strongly typed version.

Reply:
Added a ByLineElementContainer class to hold the element names